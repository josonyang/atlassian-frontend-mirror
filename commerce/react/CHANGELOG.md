# @af/commerce-react

## 0.1.3

### Patch Changes

- [`61cb5313358`](https://bitbucket.org/atlassian/atlassian-frontend/commits/61cb5313358) - Removing unused dependencies and dev dependencies

## 0.1.2

### Patch Changes

- [`9d00501a414`](https://bitbucket.org/atlassian/atlassian-frontend/commits/9d00501a414) - Ensure legacy types are published for TS 4.5-4.8

## 0.1.1

### Patch Changes

- [`41fae2c6f68`](https://bitbucket.org/atlassian/atlassian-frontend/commits/41fae2c6f68) - Upgrade Typescript from `4.5.5` to `4.9.5`

## 0.1.0

### Minor Changes

- [`56507598609`](https://bitbucket.org/atlassian/atlassian-frontend/commits/56507598609) - Skip minor dependency bump

## 0.0.5

### Patch Changes

- [`5806023831a`](https://bitbucket.org/atlassian/atlassian-frontend/commits/5806023831a) - Internal change to enforce token usage for spacing properties. There is no expected visual or behaviour change.

## 0.0.4

### Patch Changes

- [`8cc2f888c83`](https://bitbucket.org/atlassian/atlassian-frontend/commits/8cc2f888c83) - Upgrade Typescript from `4.3.5` to `4.5.5`

## 0.0.3

### Patch Changes

- [`8d4228767b0`](https://bitbucket.org/atlassian/atlassian-frontend/commits/8d4228767b0) - Upgrade Typescript from `4.2.4` to `4.3.5`.

## 0.0.2

### Patch Changes

- [`cb2392f6d33`](https://bitbucket.org/atlassian/atlassian-frontend/commits/cb2392f6d33) - Upgrade to TypeScript 4.2.4

## 0.0.1

### Patch Changes

- [`e5c858ab40c`](https://bitbucket.org/atlassian/atlassian-frontend/commits/e5c858ab40c) - Updating stripe and managing state update

## 0.0.4

### Patch Changes

- [`229b32842b5`](https://bitbucket.org/atlassian/atlassian-frontend/commits/229b32842b5) - Fix .npmignore and tsconfig.json for **tests**

## 0.0.3

### Patch Changes

- [`d3265f19be`](https://bitbucket.org/atlassian/atlassian-frontend/commits/d3265f19be) - Transpile packages using babel rather than tsc

## 0.0.2

### Patch Changes

- [`5f58283e1f`](https://bitbucket.org/atlassian/atlassian-frontend/commits/5f58283e1f) - Export types using Typescript's new "export type" syntax to satisfy Typescript's --isolatedModules compiler option.
  This requires version 3.8 of Typescript, read more about how we handle Typescript versions here: https://atlaskit.atlassian.com/get-started
  Also add `typescript` to `devDependencies` to denote version that the package was built with.

## 0.0.1

### Patch Changes

- [`b443b5a60f`](https://bitbucket.org/atlassian/atlassian-frontend/commits/b443b5a60f) - Renamed template package
