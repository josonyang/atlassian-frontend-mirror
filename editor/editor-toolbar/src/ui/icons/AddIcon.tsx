/* eslint-disable @atlaskit/editor/no-re-export */
export { default as AddIcon } from '@atlaskit/icon/core/add';
