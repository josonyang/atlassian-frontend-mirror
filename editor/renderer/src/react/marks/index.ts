import type { ComponentType } from 'react';
import type { Mark } from '@atlaskit/editor-prosemirror/model';
import Code from './code';
import Em from './em';
import Link from './link';
import Strike from './strike';
import Strong from './strong';
import Subsup from './subsup';
import TextColor from './textColor';
import BackgroundColor from './backgroundColor';
import Underline from './underline';
import Breakout from './breakout';
import Alignment from './alignment';
import Indentation from './indentation';
import UnsupportedMark from './unsupportedMark';
import UnsupportedNodeAttribute from './unsupportedNodeAttribute';
import DataConsumer from './data-consumer';
import FragmentMark from './fragment';
import Annotation, { isAnnotationMark } from './annotation';
import Border from './border';

// Stage0
import ConfluenceInlineComment from './confluence-inline-comment';

export const markToReact: {
	[key: string]: ComponentType<React.PropsWithChildren<any>>;
} = {
	code: Code,
	em: Em,
	link: Link,
	strike: Strike,
	strong: Strong,
	subsup: Subsup,
	textColor: TextColor,
	backgroundColor: BackgroundColor,
	underline: Underline,
	annotation: Annotation,
	border: Border,

	// Stage0
	confluenceInlineComment: ConfluenceInlineComment,
	breakout: Breakout,
	alignment: Alignment,
	indentation: Indentation,
	unsupportedMark: UnsupportedMark,
	unsupportedNodeAttribute: UnsupportedNodeAttribute,
	dataConsumer: DataConsumer,
	fragment: FragmentMark,
};

export const toReact = (mark: Mark): ComponentType<React.PropsWithChildren<any>> => {
	return markToReact[mark.type.name];
};

export {
	Code,
	Em,
	Link,
	Strike,
	Strong,
	Subsup,
	TextColor,
	BackgroundColor,
	Underline,
	Breakout,
	Annotation,
	Border,
	UnsupportedMark,
	isAnnotationMark,
	UnsupportedNodeAttribute,
	DataConsumer,
	FragmentMark,
};
