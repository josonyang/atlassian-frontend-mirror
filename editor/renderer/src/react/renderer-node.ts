import type { Node, Schema } from '@atlaskit/editor-prosemirror/model';

export const isLayoutNode = (node: Node) => node.type.name === 'layoutSection';

export const hasBreakOutMark = (node: Node) =>
  node.marks.some((m) => m.type.name === 'breakout');

export const insideBreakoutLayout = (path: Node[]) =>
  path.some((item) => isLayoutNode(item) && hasBreakOutMark(item));

export const insideBlockNode = (path: Node[], schema: Schema) => {
  const {
    nodes: { expand, nestedExpand, layoutColumn },
  } = schema;
  const blockNodeNames = [expand, nestedExpand, layoutColumn]
    .filter((node) => Boolean(node))
    .map((node) => node.name);

  return (
    path && path.some((n) => n.type && blockNodeNames.indexOf(n.type.name) > -1)
  );
};

export const insideMultiBodiedExtension = (path: Node[], schema: Schema) => {
  const {
    nodes: { multiBodiedExtension },
  } = schema;
  return path.some((n) => n.type === multiBodiedExtension);
};
