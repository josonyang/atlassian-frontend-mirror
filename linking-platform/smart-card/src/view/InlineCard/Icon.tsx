// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import styled from '@emotion/styled';

import { SpanSkeleton } from '@atlaskit/linking-common';
import { token } from '@atlaskit/tokens';
import { fg } from '@atlaskit/platform-feature-flags';
import React from 'react';

// TODO: Figure out a more scalable/responsive solution
// for vertical alignment.
// Current rationale: vertically positioned at the top of
// the smart card container (when set to 0). Offset this
// to position it with appropriate whitespace from the top.
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
export const Icon = styled.img({
	height: '14px',
	width: '14px',
	marginRight: token('space.050', '4px'),
	borderRadius: '2px',
	userSelect: 'none',
	position: 'absolute',
	top: '50%',
	left: '50%',
	transform: 'translate(-50%, -50%)',
});

// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
export const IconResized = styled.img({
	height: '16px',
	width: '16px',
	marginRight: token('space.050', '4px'),
	borderRadius: '2px',
	userSelect: 'none',
	position: 'absolute',
	top: '50%',
	left: '50%',
	transform: 'translate(-50%, -50%)',
});

// Used for 'untrue' icons which claim to be 16x16 but
// are less than that in height/width.
// TODO: Replace this override with proper AtlasKit solution.
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-styled, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
export const AKIconWrapper = styled.span({
	marginRight: token('space.negative.025', '-2px'),
});

export const Shimmer = ({ testId }: { testId: string }) => {
	const skeletonCustomStyles: React.CSSProperties = {
		position: 'absolute',
		// eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage/preview
		top: '50%',
		// eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage/preview
		left: '50%',
		transform: 'translate(-50%, -50%)',
		marginRight: token('space.050', '4px'),
	};

	return (
		<SpanSkeleton
			width={fg('linking-platform-increase-inline-card-icon-size') ? 16 : 14}
			height={fg('linking-platform-increase-inline-card-icon-size') ? 16 : 14}
			borderRadius={2}
			testId={testId}
			// eslint-disable-next-line @atlaskit/ui-styling-standard/enforce-style-prop -- Ignored via go/DSP-18766
			style={skeletonCustomStyles}
		/>
	);
};
