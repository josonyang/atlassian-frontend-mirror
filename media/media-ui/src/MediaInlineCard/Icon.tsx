import styled from '@emotion/styled';
import { token } from '@atlaskit/tokens';

// TODO: Figure out a more scalable/responsive solution
// for vertical alignment.
// Current rationale: vertically positioned at the top of
// the smart card container (when set to 0). Offset this
// to position it with appropriate whitespace from the top.
export const Icon = styled.img`
  height: 14px;
  width: 14px;
  margin-right: ${token('space.050', '4px')};
  border-radius: 2px;
  user-select: none;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

// Used for 'untrue' icons which claim to be 16x16 but
// are less than that in height/width.
// TODO: Replace this override with proper AtlasKit solution.
export const AKIconWrapper = styled.span`
  margin-right: ${token('space.negative.025', '-2px')};
`;
