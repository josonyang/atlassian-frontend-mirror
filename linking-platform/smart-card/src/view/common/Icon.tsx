/**
 * @jsxRuntime classic
 * @jsx jsx
 */
import React from 'react';

import { css, jsx } from '@compiled/react';
import ImageLoader from 'react-render-image';

import LinkIcon from '@atlaskit/icon/core/migration/link';

export interface IconProps {
	/* Element to be displayed as an icon if icon not provided or icon url request return error. */
	defaultIcon?: React.ReactNode;
	/* Element to be displayed as an icon. We naively render this if it is provided. Allows us to pass in AK icons */
	icon?: React.ReactNode;
	/* A prop to determine whether the icon is a Flexible UI rendered Icon, used internally by Flexible UI and Hover Preview */
	isFlexibleUi?: boolean;
	/* A `testId` prop is provided for specified elements, which is a unique string that appears as a data attribute `data-testid` in the rendered code, serving as a hook for automated tests. */
	testId?: string;
	/* Url of the icon to be displayed. Note that this is only used if a JSX element is not provided */
	url?: string;
}

/**
 * Class name for selecting non-flexible block card icon image
 *
 * @deprecated {@link https://hello.jira.atlassian.cloud/browse/ENGHEALTH-6878 Internal documentation for deprecation (no external access)}
 * Using this selctor is deprecated as once the flexible block card feature flag is removed, this class will no longer be used.
 */
export const blockCardIconImageClassName = 'block-card-icon-image';

const imageStyles = css({
	height: '16px',
	width: '16px',
});

const spanStyles = css({
	height: '20px',
	width: '16px',
	display: 'flex',
	alignItems: 'center',
	justifyContent: 'center',
});

export const Icon = ({
	url,
	icon,
	defaultIcon,
	testId = 'block-card-icon',
	isFlexibleUi = false,
}: IconProps) => {
	const placeholder = defaultIcon || (
		<LinkIcon label="link" LEGACY_size="small" testId={`${testId}-default`} color="currentColor" />
	);

	const image = url && (
		<ImageLoader
			src={url}
			loaded={
				<img css={isFlexibleUi && imageStyles} src={url} alt="" data-testid={`${testId}-image`} />
			}
			errored={placeholder}
		/>
	);

	return (
		<span
			css={isFlexibleUi && spanStyles}
			data-testid={testId}
			// eslint-disable-next-line @atlaskit/ui-styling-standard/no-classname-prop -- Ignored via go/DSP-18766
			className={blockCardIconImageClassName}
		>
			{icon || image || placeholder}
		</span>
	);
};
