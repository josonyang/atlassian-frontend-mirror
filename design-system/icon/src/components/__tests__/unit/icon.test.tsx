import React from 'react';

import { render, screen } from '@testing-library/react';

import AddIconNew from '../../../../core/add';
import AddIcon from '../../../../glyph/add';
import Icon, { type CustomGlyphProps, IconNew, type IconProps, size } from '../../../index';

describe('@atlaskit/icon', () => {
	describe('Icon', () => {
		const secretContent = 'secret content';
		const secretWrapper = ({ role }: CustomGlyphProps) => <svg role={role}>{secretContent}</svg>;
		const empty = ({ role }: CustomGlyphProps) => <svg role={role}>Icon</svg>;
		const MyIcon = (props: IconProps) => (
			// This is not ideal but is easier for testing
			// eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
			<Icon glyph={secretWrapper} {...props} />
		);

		it('should match the DOM Snapshot', () => {
			render(<Icon glyph={empty} label="My icon" />);

			expect(screen.getByRole('img')).toMatchSnapshot();
		});

		describe('glyph prop', () => {
			it('should render an SVG provided via JSX', () => {
				const id = 'customSvg';
				const customGlyphJsx = ({ role }: CustomGlyphProps) => <svg role={role} data-testid={id} />;

				render(<Icon glyph={customGlyphJsx} label="" />);
				expect(screen.getByRole('presentation', { hidden: true })).toBeInTheDocument();
			});

			it('should present itself as an image', () => {
				render(<Icon glyph={empty} testId="empty-icon" label="My icon" />);

				expect(screen.getByRole('img')).toBeInTheDocument();
			});
		});

		describe('dangerouslySetGlyph prop', () => {
			const id = 'customSvg';
			const customGlyphString = `<svg data-testid="${id}"></svg>`;

			it('should render an SVG provided as a string', () => {
				render(
					<Icon testId="test-icon" dangerouslySetGlyph={customGlyphString} label="hello-world" />,
				);
				const svg = screen.getByTestId(id);
				expect(svg).toBeInTheDocument();
				expect(svg.nodeName).toEqual('svg');
			});

			it('should present itself as an image when label is defined', () => {
				render(<Icon dangerouslySetGlyph={customGlyphString} label="hello-world" />);

				const element = screen.getByRole('img');
				expect(element).toHaveAttribute('aria-label', 'hello-world');
			});

			it('should present as hidden, without a role when the label is an empty string', () => {
				const testId = 'test-icon';
				render(<Icon testId={testId} dangerouslySetGlyph={customGlyphString} label="" />);

				const element = screen.getByTestId(testId);
				expect(element).not.toHaveAttribute('role');
				expect(element).not.toHaveAttribute('aria-label');
				expect(element).toHaveAttribute('aria-hidden', 'true');
			});
		});

		describe('exports', () => {
			it('exports handled sizes', () => {
				expect(size).not.toBe(undefined);
				expect((Object.keys(size) as (keyof typeof size)[]).map((key) => size[key])).toEqual([
					'small',
					'medium',
					'large',
					'xlarge',
				]);
			});
		});

		it('should be possible to create an Icon via a subclass', () => {
			render(<MyIcon label="My icon" />);

			expect(screen.getByRole('img')).toBeInTheDocument();
		});

		describe('primaryColor property', () => {
			const testLabel = 'test';
			it('is set to inherit the text color by default', () => {
				render(<MyIcon label={testLabel} />);

				const iconColor = screen.getByRole('img').style.getPropertyValue('--icon-primary-color');

				expect(iconColor).toEqual('currentColor');
			});
		});

		describe('secondaryColor property', () => {
			it('is set to the default theme background color by default', () => {
				render(<MyIcon label="default secondaryColor" />);

				expect(screen.getByRole('presentation')).toHaveStyle(`fill: var(--icon-secondary-color)`);
			});

			it('can be changed to a hex value', () => {
				const secondaryColor = '#ff0000';
				const label = 'hex secondaryColor';
				render(<MyIcon label={label} secondaryColor={secondaryColor} />);

				expect(screen.getByRole('presentation')).toHaveStyle(`fill: var(--icon-secondary-color)`);
			});

			it('can be changed to a named color', () => {
				const secondaryColor = 'rebeccapurple';
				const label = 'hex secondaryColor';
				render(<MyIcon label={label} secondaryColor={secondaryColor} />);

				expect(screen.getByRole('presentation')).toHaveStyle('fill: var(--icon-secondary-color)');
			});
		});
	});
	describe('Glyph', () => {
		afterEach(() => {
			jest.resetAllMocks();
		});

		const label = 'test-label';
		const glyph = <AddIcon label={label} />;

		it('should match snapshot', () => {
			render(glyph);
			expect(screen.getByRole('img')).toMatchSnapshot();
		});

		it('should have the correct label', () => {
			render(glyph);
			expect(screen.getByRole('img')).toHaveAccessibleName(label);
		});
	});
});

describe('@atlaskit/icon/base-new', () => {
	describe('dangerouslySetGlyph prop', () => {
		const id = 'customPath';
		const customPathString = `<path data-testid="${id}" d=""></path>`;

		it('should render a path provided as a string', () => {
			render(
				<IconNew testId="test-icon" dangerouslySetGlyph={customPathString} label="hello-world" />,
			);
			const path = screen.getByTestId(id);
			expect(path).toBeInTheDocument();
			expect(path.nodeName).toEqual('path');
		});

		it('should present itself as an image when label is defined', () => {
			render(<IconNew dangerouslySetGlyph={customPathString} label="hello-world" />);

			const element = screen.getByRole('img');
			expect(element).toHaveAttribute('aria-label', 'hello-world');
		});

		it('should present as hidden, without a role when the label is an empty string', () => {
			const testId = 'test-icon';
			render(<IconNew testId={testId} dangerouslySetGlyph={customPathString} label="" />);

			const element = screen.getByTestId(testId);
			expect(element).not.toHaveAttribute('role');
			expect(element).not.toHaveAttribute('aria-label');
			expect(element).toHaveAttribute('aria-hidden', 'true');
		});
	});
});

describe('Glyph', () => {
	afterEach(() => {
		jest.resetAllMocks();
	});

	const label = 'test-label';
	const glyph = <AddIconNew label={label} />;

	it('should match snapshot', () => {
		render(glyph);
		expect(screen.getByRole('img')).toMatchSnapshot();
	});

	it('should have the correct label', () => {
		render(glyph);
		expect(screen.getByRole('img')).toHaveAccessibleName(label);
	});
});
