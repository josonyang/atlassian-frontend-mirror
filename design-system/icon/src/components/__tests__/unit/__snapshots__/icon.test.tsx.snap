// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@atlaskit/icon Glyph should match snapshot 1`] = `
.emotion-0 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
}

.emotion-0 >svg {
  overflow: hidden;
  pointer-events: none;
  max-width: 100%;
  max-height: 100%;
  color: var(--icon-primary-color);
  fill: var(--icon-secondary-color);
  vertical-align: bottom;
}

.emotion-0 >svg stop {
  stop-color: currentColor;
}

@media screen and (forced-colors: active) {
  .emotion-0 >svg {
    -webkit-filter: grayscale(1);
    filter: grayscale(1);
    --icon-primary-color: CanvasText;
    --icon-secondary-color: Canvas;
  }
}

<span
  aria-label="test-label"
  class="emotion-0"
  role="img"
  style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
>
  <svg
    height="24"
    role="presentation"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M13 11V3.993A.997.997 0 0 0 12 3c-.556 0-1 .445-1 .993V11H3.993A.997.997 0 0 0 3 12c0 .557.445 1 .993 1H11v7.007c0 .548.448.993 1 .993.556 0 1-.445 1-.993V13h7.007A.997.997 0 0 0 21 12c0-.556-.445-1-.993-1z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</span>
`;

exports[`@atlaskit/icon Icon should match the DOM Snapshot 1`] = `
.emotion-0 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
}

.emotion-0 >svg {
  overflow: hidden;
  pointer-events: none;
  max-width: 100%;
  max-height: 100%;
  color: var(--icon-primary-color);
  fill: var(--icon-secondary-color);
  vertical-align: bottom;
}

.emotion-0 >svg stop {
  stop-color: currentColor;
}

@media screen and (forced-colors: active) {
  .emotion-0 >svg {
    -webkit-filter: grayscale(1);
    filter: grayscale(1);
    --icon-primary-color: CanvasText;
    --icon-secondary-color: Canvas;
  }
}

<span
  aria-label="My icon"
  class="emotion-0"
  role="img"
  style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
>
  <svg
    role="presentation"
  >
    Icon
  </svg>
</span>
`;

exports[`Glyph should match snapshot 1`] = `
.emotion-1 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
  -webkit-padding-end: var(--ds--button--new-icon-padding-end, 0);
  padding-inline-end: var(--ds--button--new-icon-padding-end, 0);
  -webkit-padding-start: var(--ds--button--new-icon-padding-start, 0);
  padding-inline-start: var(--ds--button--new-icon-padding-start, 0);
}

@media screen and (forced-colors: active) {
  .emotion-1 {
    color: CanvasText;
    -webkit-filter: grayscale(1);
    filter: grayscale(1);
  }
}

.emotion-0 {
  overflow: hidden;
  pointer-events: none;
  color: currentColor;
  vertical-align: bottom;
  width: var(--ds-space-200, 16px);
  height: var(--ds-space-200, 16px);
}

.emotion-0 stop {
  stop-color: currentColor;
}

<span
  aria-label="test-label"
  class="emotion-1"
  role="img"
>
  <svg
    class="emotion-0"
    fill="none"
    role="presentation"
    viewBox="0 0 16 16"
  >
    <path
      d="M8 8h7M8 8H1m7 0V1m0 7v7"
      stroke="currentColor"
      stroke-width="1.5"
    />
  </svg>
</span>
`;
